apps/ui/src/Components/Map/Map.tsx

```diff
@@ -1,81 +1,113 @@
-import { Icon, divIcon, point } from 'leaflet';
+
 import placeholderIcon from '../../assets/components/map/car.svg';
-import { GeoSearchControl, OpenStreetMapProvider } from 'leaflet-geosearch';
 
 type MarkerInfo = {
-    geocode: [number, number];
-    popUp: string;
+  geocode: [number, number];
+  popUp: string;
 };
 
+interface MapProps {
+  focusLocation?: [number, number];
+  markers: MarkerInfo[];
+}
+
 const customIcon = new (Icon as any)({
-    iconUrl: placeholderIcon,
-    iconSize: [38, 38]
+  iconUrl: placeholderIcon,
+  iconSize: [38, 38],
 });
 
-const createClusterCustomIcon = (cluster: any) => new (divIcon as any)({
+const createClusterCustomIcon = (cluster: any) =>
+  new (divIcon as any)({
     html: `<span class="cluster-icon">${cluster.getChildCount()}</span>`,
-    className: "custom-marker-cluster",
-    iconSize: point(33, 33, true)
-});
-
-const markers: MarkerInfo[] = [
-    // Your markers
-];
+    className: 'custom-marker-cluster',
+    iconSize: point(33, 33, true),
+  });
 
 const SearchControl = () => {
-    const map = useMap();
-    React.useEffect(() => {
-        const provider = new OpenStreetMapProvider();
-
-        const searchControl = new (GeoSearchControl as any)({
-            provider,
-            style: 'bar',
-            autoComplete: true,
-            autoCompleteDelay: 250,
-            showMarker: true,
-            showPopup: false,
-            marker: {
-                icon: new (Icon as any)({
-                    iconUrl: placeholderIcon,
-                    iconSize: [25, 41],
-                    iconAnchor: [12, 41]
-                })
-            }
-        });
-
-        map.addControl(searchControl);
-        return () => {
-            map.removeControl(searchControl);
-        };
-    }, [map]);
-
-    return null;
+  const map = useMap();
+  useEffect(() => {
+    const provider = new OpenStreetMapProvider();
+
+    const searchControl = new (GeoSearchControl as any)({
+      provider,
+      style: 'bar',
+      autoComplete: true,
+      autoCompleteDelay: 250,
+      showMarker: true,
+      showPopup: false,
+      marker: {
+        icon: new (Icon as any)({
+          iconUrl: placeholderIcon,
+          iconSize: [25, 41],
+          iconAnchor: [12, 41],
+        }),
+      },
+    });
+
+    map.addControl(searchControl);
+    return () => {
+      map.removeControl(searchControl);
+    };
+  }, [map]);
+
+  return null;
 };
 
-const Map: React.FC = () => (
+const Focus = (props: { focusLocation: [number, number] }) => {
+  const map = useMap();
+  const { focusLocation } = props;
+
+  useEffect(() => {
+    if (focusLocation) map.flyTo(focusLocation, 18);
+  }, [focusLocation]);
+
+  return null;
+};
+
+const Map: React.FC<MapProps> = ({ markers, focusLocation }) => {
+  const [focusedLocation, setFocusedLocation] = useState<
+    [number, number] | undefined
+  >(focusLocation);
+
+  useEffect(() => {
+    setFocusedLocation(focusLocation);
+  }, [focusLocation]);
```

Comment:
is this neccesery? doesnt do this anyway?

Reply:
No, the state does not reset when the props change